float32[3] gains
---
